### DISCUSSION TITLE
Custom reusable components in form and outside

### DISCUSSION THREAD

**Discussion by RobinBobo** [react-hook-form/react-hook-form#9797] [Q&A] [answered]
Hi,
I want to make a component usable with `react-hook-form` but also outside of it.
For that i used `useController` like this : 

```
import { Select as MantineSelect, SelectProps as MantineSelectProps } from '@mantine/core'
import React from 'react'
import { Control, FieldValues, useController, UseControllerProps } from 'react-hook-form'

interface SelectProps<T extends FieldValues>
  extends UseControllerProps<T>,
    Omit<MantineSelectProps, 'defaultValue' | 'name' | 'value'> {
  control: Control<T>
}

export const Select = <T extends FieldValues>({
  control,
  name,
  rules,
  shouldUnregister,
  defaultValue,
  ...props
}: SelectProps<T>) => {
  const { field } = useController({
    name,
    control,
    defaultValue,
    shouldUnregister,
    rules,
  })

  return (
    <MantineSelect
      ref={field.ref}
      name={name}
      value={field.value}
      defaultValue={defaultValue}
      onBlur={field.onBlur}
      onChange={field.onChange}
      {...props}
    />
  )
}
```

The only problem is that if i don't pass `control` to the component, then the hook `useController` breaks with `medthod is null`.

Is there any solution wihtout having to do a wrapping component that redirects to the normal component or the enhanced one for `react-hook-form` depending on the `control` value ?

(I don't want to use wrapper `Controller` that i find too wordy or even with a const register, and manually add value, onChange, onBlur properties, if possible)

**Comment by martin-wahlberg** [comment, marked as answer]
Could you make a codesandbox with the intended use of the component?

**Reply by RobinBobo** [reply]
Here you go :) 
https://codesandbox.io/s/rhf-custom-reusable-components-in-form-and-outside-o16x8y
I used a custom select in the sandbox, but it's the same intended use with any other components

**Reply by martin-wahlberg** [reply]
Yeah ok! Then i understand your issue. It is not possible without breaking the [rules of hooks](https://reactjs.org/docs/hooks-rules.html#:~:text=Don%E2%80%99t%20call%20Hooks,depth%20below.). So i think you probably would be better of by splitting the  component into a controlled and a standalone variant. You could probably do it with the Controller component as well but i imagine that only would add complexity.

**Reply by RobinBobo** [reply]
Okay thanks for your anwser !
This confirm what I thought
